### PR TITLE
test: Verify that skipping versions fails

### PR DESCRIPTION
**Pull request by jkosh44**
This commit adds a test to assert that upgrading by skipping versions will fail.

Works toward resolving #24218

### Motivation
Adds a test.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

File: test/persist-catalog-migration/mzcompose.py [hunk 1]
```diff
@@ -401,3 +405,43 @@ def check_objects(c: Composition, i: int):
     """
         )
     )
+
+
+def workflow_test_version_skips(c: Composition) -> None:
```
Comment (jkosh44) [inline test/persist-catalog-migration/mzcompose.py:410]: I'm not sure if this is the best file for this, but I didn't know where else to put it.

File: misc/python/materialize/cli/ci_logged_errors_detect.py [hunk 2]
```diff
@@ -113,6 +113,8 @@
     | internal\ error:\ no\ AWS\ external\ ID\ prefix\ configured
     # For persist-catalog-migration ignore failpoint panics
     | persist-catalog-migration-materialized.* \| .* failpoint\ .* panic
+    # For tests we purposely trigger this error
+    | persist-catalog-migration-materialized.* \| .* incompatible persist version \d+\.\d+\.\d+(-dev)?, current: \d+\.\d+\.\d+(-dev)?, make sure to upgrade the catalog one version at a time
```
Comment (jkosh44) [inline misc/python/materialize/cli/ci_logged_errors_detect.py:117]: This doesn't seem to be working and the test is still failing due to this log, any advice?

Comment (nrainer-materialize) [inline misc/python/materialize/cli/ci_logged_errors_detect.py:117]:
```suggestion
    | persist-catalog-migration-materialized.* \| .* incompatible\ persist\ version\ \d+\.\d+\.\d+(-dev)?,\ current:\ \d+\.\d+\.\d+(-dev)?,\ make\ sure\ to\ upgrade\ the\ catalog\ one\ version\ at\ a\ time
```

Comment (nrainer-materialize) [reply misc/python/materialize/cli/ci_logged_errors_detect.py:117]: This is due to `re.VERBOSE`.

**Review by nrainer-materialize — APPROVED** [review]
Thank you for the test!

**Pull request merged**